ktor-client/ktor-client-curl/desktop/src/io/ktor/client/engine/curl/internal/CurlWebSocketSession.kt

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2014-2024 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.engine.curl.internal
+
+import io.ktor.utils.io.core.*
+import io.ktor.websocket.*
+import kotlinx.atomicfu.*
+import kotlinx.cinterop.*
+import kotlinx.coroutines.*
+import kotlinx.coroutines.channels.*
+import libcurl.*
+import kotlin.coroutines.*
+
+@OptIn(ExperimentalForeignApi::class)
+internal class CurlWebSocketSession(
+    private val websocket: CurlWebSocketResponseBody,
+    callContext: CoroutineContext
+) : WebSocketSession, Closeable {
+
+    private val closed = atomic(false)
+    private val socketJob = Job(callContext[Job])
+    private val _outgoing = Channel<Frame>(Channel.UNLIMITED)
+
+    override val coroutineContext: CoroutineContext = callContext + socketJob + CoroutineName("curl-ws")
+    override var masking: Boolean
+        get() = true
+        set(_) {}
```

Comment:
This unimplemented stuff makes me think there's too many things in the `WebSocketSession` interface... I see there's other implementations with similar declarations.